src/ac_copilot_trainer/modules/lap_archive.lua

```
@@ -0,0 +1,340 @@
+---  opts.session_uuid (string), opts.car (StateCar|nil), opts.sim (StateSim|nil),
+---  opts.lap_n (int), opts.lap_ms (int), opts.is_pb (bool), opts.is_valid (bool),
+---  opts.trace (per-sample objects), opts.corners (corner_features list),
+---  opts.setup_snap (setupReader snap), opts.setup_hash (string),
+---  opts.rules_hints (string list), opts.sidecar_debrief (string|nil),
+---  opts.corner_advice (table label->text|nil)
+---@return table|nil
+function M.buildRecord(opts)
+  if type(opts) ~= "table" then return nil end
+  if not opts.lap_n or not opts.lap_ms or opts.lap_ms <= 0 then return nil end
+
+  local car = opts.car
```

Luacheck (1.2.0)

[warning] unused variable 'car'

(W211)